components/pixels/snapshot.rs

```diff
@@ -180,31 +195,25 @@ impl Snapshot<SnapshotData> {
         }
     }
 
-    pub fn get_rect(&self, rect: Rect<u32>) -> Self {
-        let data = rgba8_get_rect(self.as_raw_bytes(), self.size(), rect).to_vec();
-        Self::from_vec(rect.size, self.format, self.alpha_mode, data)
-    }
-
-    // TODO: https://github.com/servo/servo/issues/36594
-    /*
-    /// # Safety
-    ///
-    /// This is safe if data is owned by this process only
-    /// (ownership is transferred on send)
-    pub unsafe fn from_shared_memory(
+    pub fn from_shared_memory(
         size: Size2D<u32>,
-        format: PixelFormat,
-        alpha_mode: AlphaMode,
-        ism: IpcSharedMemory,
+        format: SnapshotPixelFormat,
+        alpha_mode: SnapshotAlphaMode,
+        data: Arc<IpcSharedMemory>,
+        byte_range: Range<usize>,
```

Comment:
could we accept https://doc.rust-lang.org/nightly/core/ops/trait.RangeBounds.html (and then map .. to 0..size)

Reply:
Sure. I've done that.